python/src/equistore/operations/slice.py

```diff
@@ -358,35 +280,42 @@ def _slice_block(block: TensorBlock, samples=None, properties=None) -> TensorBlo
     return new_block
 
 
-def _slice_tensormap(tensormap: TensorMap, samples=None, properties=None) -> TensorMap:
+def _check_args(
+    tensor: Union[TensorMap, TensorMap],
+    samples: Optional[Labels] = None,
+    properties: Optional[Labels] = None,
+):
     """
-    Slices an input :py:class:`TensorMap` along the samples and/or properties
-    dimension(s). ``samples`` and ``properties`` are
-    :py:class:`Labels` objects that specify the samples/properties
-    (respectively) names and indices that should be sliced, i.e. kept in the
-    output tensor.
-
-    Note that either ``samples`` or ``properties``, or both,
-    should be specified as input.
-
-    :param tensor: the input :py:class:`TensorMap` to be sliced.
-    :param samples: a :py:class:`Labels` object containing the names
-        and indices of samples to keep in the sliced :py:class:`TensorBlock`
-        output, or each of the sliced :py:class:`TensorBlock` objects of the
-        output :py:class:`TensorMap`. Default value of None indicates no slicing
-        along the samples dimension should occur.
-    :param properties: a :py:class:`Labels` object containing the names
-        and indices of properties to keep in the sliced :py:class:`TensorBlock`
-        output, or each of the sliced :py:class:`TensorBlock` objects of the
-        output :py:class:`TensorMap`. Default value of None indicates no slicing
-        along the properties dimension should occur.
-
-    :return: a :py:class:`TensorMap` that corresponds to the sliced input
-        tensor.
+    Checks the arguments passed to :py:func:`slice` and :py:func:`slice_block`.
     """
-
-    # Iterate over, and slice, each block (+ gradients) of the input TensorMap in turn.
-    return TensorMap(
-        keys=tensormap.keys,
-        blocks=[_slice_block(block, samples, properties) for _, block in tensormap],
-    )
+    # Must pass either samples or properties or both
+    if samples is None and properties is None:
+        raise ValueError(
+            "you must specify either samples or properties (or both) to slice by"
+        )
```

Comment:
What is happening if we don't have this check? Are the slice and split function only returning a copy of the original Block? If yes we might think about removing this test.

Reply:
Currently, yes if we take out this function ``slice(tensor, samples=None, properties=None)`` will return a copy of the input tensor. Is this the behaviour we want? Or do we want to throw an error if neither samples or properties are specified?

Reply:
Similarly if I call ``split(tensor, axis="samples", grouped_idxs=[])`` should this also return the original tensor, or throw an error?

@DavideTisi thoughts?